cirq/pasqal/pasqal_qubits_test.py

```diff
@@ -0,0 +1,249 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pytest
+import numpy as np
+
+import cirq
+from cirq.pasqal import ThreeDGridQubit
+
+
+def test_pasqal_qubit_init():
+    q = ThreeDGridQubit(3, 4, 5)
+    assert q.row == 3
+    assert q.col == 4
+    assert q.lay == 5
+
+
+def test_comparison_key():
+    assert ThreeDGridQubit(3, 4, 5)._comparison_key() == (3, 4, 5)
+
+
+def test_grid_qubit_eq():
+    eq = cirq.testing.EqualsTester()
+    eq.make_equality_group(lambda: ThreeDGridQubit(0, 0, 0))
+    eq.make_equality_group(lambda: ThreeDGridQubit(1, 0, 0))
+    eq.make_equality_group(lambda: ThreeDGridQubit(0, 1, 0))
+    eq.make_equality_group(lambda: ThreeDGridQubit(50, 25, 25))
+
+
+def test_square():
+    assert ThreeDGridQubit.square(2, top=1, left=1) == [
+        ThreeDGridQubit(1, 1, 0),
+        ThreeDGridQubit(1, 2, 0),
+        ThreeDGridQubit(2, 1, 0),
+        ThreeDGridQubit(2, 2, 0)
+    ]
+    assert ThreeDGridQubit.square(2) == [
+        ThreeDGridQubit(0, 0, 0),
+        ThreeDGridQubit(0, 1, 0),
+        ThreeDGridQubit(1, 0, 0),
+        ThreeDGridQubit(1, 1, 0)
+    ]
+
+
+def test_rec():
+    assert ThreeDGridQubit.rect(1, 2, top=5, left=6) == [
+        ThreeDGridQubit(5, 6, 0),
+        ThreeDGridQubit(5, 7, 0)
+    ]
+    assert ThreeDGridQubit.rect(2, 2) == [
+        ThreeDGridQubit(0, 0, 0),
+        ThreeDGridQubit(0, 1, 0),
+        ThreeDGridQubit(1, 0, 0),
+        ThreeDGridQubit(1, 1, 0)
+    ]
+
+
+def test_cube():
+    assert ThreeDGridQubit.cube(2, top=1, left=1, upper=1) == [
+        ThreeDGridQubit(1, 1, 1),
+        ThreeDGridQubit(1, 1, 2),
+        ThreeDGridQubit(1, 2, 1),
+        ThreeDGridQubit(1, 2, 2),
+        ThreeDGridQubit(2, 1, 1),
+        ThreeDGridQubit(2, 1, 2),
+        ThreeDGridQubit(2, 2, 1),
+        ThreeDGridQubit(2, 2, 2)
+    ]
+    assert ThreeDGridQubit.cube(2) == [
+        ThreeDGridQubit(0, 0, 0),
+        ThreeDGridQubit(0, 0, 1),
+        ThreeDGridQubit(0, 1, 0),
+        ThreeDGridQubit(0, 1, 1),
+        ThreeDGridQubit(1, 0, 0),
+        ThreeDGridQubit(1, 0, 1),
+        ThreeDGridQubit(1, 1, 0),
+        ThreeDGridQubit(1, 1, 1),
+    ]
+
+
+def test_parrallelep():
+    assert ThreeDGridQubit.parallelep(1, 2, 2, top=5, left=6, upper=7) == [
+        ThreeDGridQubit(5, 6, 7),
+        ThreeDGridQubit(5, 6, 8),
+        ThreeDGridQubit(5, 7, 7),
+        ThreeDGridQubit(5, 7, 8),
+    ]
+
+    assert ThreeDGridQubit.parallelep(2, 2, 2) == [
+        ThreeDGridQubit(0, 0, 0),
+        ThreeDGridQubit(0, 0, 1),
+        ThreeDGridQubit(0, 1, 0),
+        ThreeDGridQubit(0, 1, 1),
+        ThreeDGridQubit(1, 0, 0),
+        ThreeDGridQubit(1, 0, 1),
+        ThreeDGridQubit(1, 1, 0),
+        ThreeDGridQubit(1, 1, 1)
+    ]
+
+
+def test_pasqal_qubit_ordering():
+    assert ThreeDGridQubit(0, 0, 0) < ThreeDGridQubit(0, 0, 1)
```

Comment:
It seems like we are iterating through all `2^n` bits where `n = 3` Let's use a loop instead

```
for i in 8:
    assert ThreeDGridQubit(0, 0, 0) < ThreeDGridQubit(i)
```

Something of that nature.